### PR TITLE
Fixed cast for primary key in PrestaShopCollection

### LINKED ISSUE #20452: BO - Invoice / Product taxes detail is not displayed on the invoice depending on where the invoice is called
#### Describe the bug

For the same order, depending on where the invoice is called ; the products taxes details is not displayed / printed on the invoice 😱 

#### Expected behavior

The product taxes details should be printed / displayed on the invoice no matter where the invoice is called from.

#### Steps to Reproduce

1. Go to FO and create an order
2. Go to the BO => go to the detail of the previously created order
3. Generate the invoice for this order
4. Consult the invoice => the product taxes detail is on the invoice
5. Go back to the order list
6. Click on the quick view to see the freshly generated invoice
7. See error => the product taxes detail is not on the invoice

**Screenshots**

**Invoice from the order detail**
Screenshot at 15-43-32

**Invoice from the orders list**
Screenshot at 15-43-59

#### Additional information

* PrestaShop version: 1.7.7.x
* PHP version: 7.1

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Fixed cast for primary key in PrestaShopCollection
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #20452
| How to test?  | Cf.#20452

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20477)

### CODE REVIEW

File: classes/ObjectModel.php [hunk 1]
```diff
@@ -1927,7 +1927,7 @@ public function hydrate(array $data, $id_lang = null)
     {
         $this->id_lang = $id_lang;
         if (isset($data[$this->def['primary']])) {
-            $this->id = $data[$this->def['primary']];
+            $this->id = (int) $data[$this->def['primary']];
```
Comment (matks) [inline classes/ObjectModel.php:1930]: It was this line the responsible ? Woha 😄  nice catch

**Review by matks — DISMISSED** [review]
Code looks good. I dont think QA can "test all ObjectModels work correctly" however I believe something so impactful as this change should raise issues in our automated tests if there is one problem

File: classes/ObjectModel.php (hunk 1 above)
Comment (Progi1984) [reply classes/ObjectModel.php:1930]: Yes !

**Review by matks — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by SD1982** [comment]
LGTM Thanks @Progi1984 !!

**Pull request merged**